src/components/user-group/repo/classes.repo.integration.test.js

```diff
@@ -0,0 +1,137 @@
+		it('when the function is called with user id, it should return list of classes where the student is part of ', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const testClass = await testObjects.createTestClass({ userIds: [student._id], schoolId });
+			const classes = await classesRepo.getClassesForStudent(student._id);
+
+			expect(classes[0]._id).to.deep.equal(testClass._id);
+			expect(classes[0].userIds[0]).to.deep.equal(student._id);
+		});
+
+		it("when function is called with userId for which classes don't exist, it should return empty array", async () => {
+			const student = await testObjects.createTestUser({ roles: ['student'] });
+			const classes = await classesRepo.getClassesForStudent(student._id);
+			expect(classes.length).to.be.equal(0);
+		});
+	});
+
+	describe('get classes for teacher', async () => {
+		it('when the function is called with teacher id, it should return list of classes where the teacher is part of ', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'], schoolId });
+			const testClass = await testObjects.createTestClass({ teacherIds: [teacher._id], schoolId });
+			const classes = await classesRepo.getClassesForTeacher(teacher._id);
+
+			expect(classes[0]._id).to.deep.equal(testClass._id);
+			expect(classes[0].teacherIds[0]).to.deep.equal(teacher._id);
+		});
+
+		it("when function is called with teacherId for which classes don't exist, it should return empty array", async () => {
+			const teacher = await testObjects.createTestUser({ roles: ['teacher'] });
+			const classes = await classesRepo.getClassesForTeacher(teacher._id);
+			expect(classes.length).to.be.equal(0);
+		});
+	});
+
+	describe('delete student from classes', () => {
+		it('when the function is called, it should return list of updated classes', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const testClass = await testObjects.createTestClass({ userIds: [student._id], schoolId });
+
+			const result = await classesRepo.removeStudentFromClasses(student._id, [testClass._id]);
+
+			expect(result.length).to.be.equal(1);
+			expect(result[0]).to.be.equal(testClass._id);
+		});
+
+		it('when the student is deleted from classes, the class should not contain this student any more', async () => {
+			const { _id: schoolId } = await testObjects.createTestSchool();
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId });
+			const testClass = await testObjects.createTestClass({ userIds: [student._id], schoolId });
+
+			await classesRepo.removeStudentFromClasses(student._id, [testClass._id]);
+			const result = await classModel
```

Comment:
can't we get classes using the repository here?

Reply:
We can, but the notice from @Metauriel was that we should use native functionality instead our functions in order to isolate the testing method from the other code.

Reply:
Yes, it is a drawback  - but  **IMHO**, using the model directly is even worse - we depend on the db structure then (because of the queries) and make the test fragile (because of the queries written as JSON)
We should try to avoid super-uc-specific Repo methods (that is true). However we should have and use the generic one e.g. findById, findByIds or findByName etc instead.